helm/kyverno-policies-observability/templates/PodMonitor.yaml

```diff
@@ -0,0 +1,40 @@
+# THIS FILE IS GENERATED WITH 'make generate' - DO NOT EDIT MANUALLY
+apiVersion: kyverno.io/v1
+kind: ClusterPolicy
+metadata:
+  name: configure-pod-monitor-labelling-schema
+spec:
+  rules:
+    - name: configure-pod-monitor-labelling-schema
+      match:
+        resources:
+          kinds:
+          - monitoring.coreos.com/v1/PodMonitor
+      mutate:
+        patchStrategicMerge:
+          spec:
+            podMetricsEndpoints:
+              - relabelings:
+                  # Pod Monitors are currently only supported on management clusters so we can enforce the cluster_id and cluster_type.
+                - replacement: "{{ .Values.managementCluster.name }}"
+                  targetLabel: cluster_id
+                - replacement: "management_cluster"
+                  targetLabel: cluster_type
+                - replacement: "{{ .Values.provider.kind }}"
+                  targetLabel: provider
+                - replacement: "{{ .Values.managementCluster.name }}"
+                  targetLabel: installation
+                - sourceLabels: [__meta_kubernetes_namespace]
+                  targetLabel: namespace
+                - sourceLabels: [__meta_kubernetes_pod_name]
+                  targetLabel: pod
+                - sourceLabels: [__meta_kubernetes_pod_container_name]
+                  targetLabel: container
+                - sourceLabels: [__meta_kubernetes_pod_node_name]
+                  targetLabel: node
+                - sourceLabels: [__meta_kubernetes_node_label_role]
+                  targetLabel: role
+                - sourceLabels: [__meta_kubernetes_node_label_role]
+                  regex: ""
```

Comment:
What's this empty regex for?

Reply:
We take the role from the __meta_kubernetes_node_label_role label but if it is empty we set it to worker by default. We set that in the scrape config also

Reply:
But you are right this workaround was removed https://github.com/giantswarm/prometheus-meta-operator/blob/master/files/templates/scrapeconfigs/_node_role.yaml